pkg/uefi/meregion.go

```diff
@@ -95,15 +95,20 @@ func (e MEPartitionEntry) Type() string {
 
 // FindMEDescriptor searches for an Intel ME FTP signature
 func FindMEDescriptor(buf []byte) (int, error) {
-	if bytes.Equal(buf[16:16+len(MEFTPSignature)], MEFTPSignature) {
+	if bytes.Equal(buf[16:16+len(MEFPTSignature)], MEFPTSignature) {
 		// 16 + 4 since the descriptor starts after the signature
```

Comment:
why not just call bytes.Index? 

Also, so many people depend on this, it needs to come with a test at this point (you found a bug! bad! bad!) :-)

Reply:
I'm just extending the cases here. Since there is no public spec to my knowledge (I haven't found any) it's hard to say what would be correct. All this knowledge is based on RE.

There are cases where the assumption that the signature is either at the start or in the second 16 bytes holds, and me_cleaner does the same:
https://chromium.googlesource.com/chromiumos/third_party/coreboot/+/refs/heads/stabilize-12239.46.B/util/me_cleaner/me_cleaner.py#579

Looking at more and newer images though, other cases show up, so MEAnalyzer tries a bunch of variants as well, with different implications:
https://github.com/platomav/MEAnalyzer/blob/6d898a6d3972ccec53099e05c5e84ec2abe79197/MEA.py#L11604

So yea bug or... case unknown up until now.
Would you just list the cases known so far for the test cases? I'll see what I can do.

Reply:
Test added. Tbh I don't see the real value, since I am now testing `bytes.Index()`... Would rather add some fixtures and use those instead. Then I'd be unsure because of ME binary distribution issues.